src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -254,17 +260,27 @@ public override OneOrMany<SyntaxList<AttributeListSyntax>> GetAttributeDeclarati
                             throw ExceptionUtilities.UnexpectedValue(accessor.Kind());
                     }
 
-                    if (accessor.Body != null || accessor.ExpressionBody != null)
+                    var body = (SyntaxNode?)accessor.Body ?? accessor.ExpressionBody;
+                    if (body != null)
                     {
                         accessorsHaveImplementation = true;
                     }
+
+                    usesFieldKeyword = usesFieldKeyword || containsFieldKeyword(accessor);
                 }
             }
             else
             {
-                accessorsHaveImplementation = GetArrowExpression(syntax) is object;
+                var body = GetArrowExpression(syntax);
+                accessorsHaveImplementation = body is object;
+                usesFieldKeyword = body is { } && containsFieldKeyword(body);
                 Debug.Assert(accessorsHaveImplementation); // it's not clear how this even parsed as a property if it has no accessor list and no arrow expression.
             }
+
+            static bool containsFieldKeyword(SyntaxNode syntax)
+            {
+                return syntax.DescendantTokens().Any(static t => t.Kind() == SyntaxKind.FieldKeyword);
```

Comment:
Does this traversal avoid creating the red tree?

Reply:
No, but haven't we already created the red tree for the property simply by referencing the root `SyntaxNode`?

Reply:
no.  walking down the red tree causes the red allocations.  ideally this walk is done on the green tree to avoid actual allocs.